test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -365,9 +329,42 @@ public static void assertSearchHit(SearchResponse searchResponse, int number, Ma
     }
 
     public static void assertNoFailures(SearchRequestBuilder searchRequestBuilder) {
+        assertNoFailuresAndResponse(searchRequestBuilder, r -> {});
+    }
+
+    public static void assertNoFailuresAndResponse(SearchRequestBuilder searchRequestBuilder, Consumer<SearchResponse> consumer) {
```

Comment:
If we're doing this (and I think it's fine) let's do a `CheckedConsumer` here and allow any exception maybe? Otherwise the code might get even more verbose if we have to deal with anything that throws checked?

Reply:
I am not sure adding a `CheckedConsumer` here as it will have  huge effect everywhere. I hope we will understand what we need here as we add more cases so I would prefer to keep it simple until we know better what we need.

Reply:
Nevermind, sorry just waking up fully seemingly :D We can change that later if need be ... my 🧠 thought we couldn't earlier :)